### PR TITLE
[NGRM] - Webviz map component - Z values should be defined geological…

### PR DESCRIPTION
**Pull request by nilscb**
…y #1150

### CODE REVIEW

File: react/src/lib/components/DeckGLMap/layers/axes/axesLayer.ts [hunk 1]
```diff
@@ -11,13 +11,17 @@ import BoxLayer from "./boxLayer";
...
 
 export interface AxesLayerProps<D> extends ExtendedLayerProps<D> {
     bounds: [number, number, number, number, number, number];
     labelColor?: Color;
     labelFontSize?: number;
     fontFamily?: string;
     axisColor?: Color;
+    // If true means that input z values are interpreted as depths.
+    // For example depth of z = 1000 corresponds to -1000 on the z axis. Default false.
+    isZDepht: boolean;
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/axes/axesLayer.ts:24]: Should be spelled `isZDepth`

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts [hunk 2]
```diff
@@ -90,14 +91,15 @@ export const layersDefaultProps: Record<string, unknown> = {
         outline: true,
         logRadius: 10,
         logCurves: true,
-        refine: true,
+        refine: false,
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts:94]: Why this change?

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts [hunk 3]
```diff
@@ -55,6 +55,7 @@ export const layersDefaultProps: Record<string, unknown> = {
         smoothShading: true,
         material: true,
         depthTest: true,
+        isZDepht: true,
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts:58]: This mechanism for setting defaults was AFAIK added to support exposing layer settings in the toolbox. It is no longer needed and adds complexity.

We should consider applying default values according to convention. Example: https://gist.githubusercontent.com/gfox1984/d0360f921b5629dad7901e7527bd2cf2/raw/960a2d1a890503f99385442133d03d3c6aad5ecd/medium-default-props-simple-comp.ts

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts (hunk 2 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts:94]: The wells are extremely slow using refine and there are no visible change when turning it of

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts (hunk 3 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts]: This may be handled in a different pull requesst?

File: react/src/lib/components/DeckGLMap/layers/axes/axesLayer.ts (hunk 1 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/axes/axesLayer.ts]: Fixed

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts [hunk 4]
```diff
@@ -215,13 +222,18 @@ export interface MapLayerProps<D> extends ExtendedLayerProps<D> {
 
     // Enable/disable depth testing when rendering layer. Default true.
     depthTest: boolean;
+
+    // If true means that input z values are interpreted as depths.
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts:226]: Please consider proper tsdoc format

File: react/src/lib/components/DeckGLMap/layers/grid3d/grid3dLayer.ts [hunk 5]
```diff
@@ -36,6 +36,13 @@ function GetBBox(
     return [xmin, ymin, zmin, xmax, ymax, zmax];
 }
 
+// XXX
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/grid3d/grid3dLayer.ts:39]: Please remove

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts (hunk 3 above)
Comment (hkfb) [reply react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts]: Referenced in https://github.com/equinor/webviz-subsurface-components/issues/1395

File: react/src/lib/components/DeckGLMap/layers/grid3d/grid3dLayer.ts (hunk 5 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/grid3d/grid3dLayer.ts]: Fixed

File: react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts (hunk 4 above)
Comment (nilscb) [reply react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts]: Fixed

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts [hunk 6]
```diff
@@ -123,6 +125,7 @@ export const layersDefaultProps: Record<string, unknown> = {
         name: "Axes",
         id: "axes-layer",
         visible: true,
+        isZDepht: false,
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts:128]: Spelling again...

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/SubsurfaceViewer/layers/wells/wellsLayer.ts [hunk 7]
```diff
@@ -298,10 +302,14 @@ export default class WellsLayer extends CompositeLayer<
         }
 
         const refine = this.props.refine;
-        const data = refine
+        let data = refine
             ? splineRefine(this.props.data as unknown as FeatureCollection) // smooth well paths.
             : (this.props.data as unknown as FeatureCollection);
 
+        if (!this.props.isZDepth) {
```
Comment (hkfb) [inline react/src/lib/components/SubsurfaceViewer/layers/wells/wellsLayer.ts:309]: Not sure if it makes any different, but maybe it's more efficient to invert before refining (less samples to invert)?

**Review by hkfb — APPROVED** [review]
